File: core/src/bara/game/util/ComponentLookup.java
package bara.game.util;

import com.badlogic.ashley.core.ComponentMapper;
import bara.game.components.*;
import com.badlogic.ashley.core.Entity;

public class ComponentLookup {
    private static ComponentMapper<BehaviorComponent> _behavior = ComponentMapper.getFor(BehaviorComponent.class);
    private static ComponentMapper<CameraComponent> _camera = ComponentMapper.getFor(CameraComponent.class);
    private static ComponentMapper<ChildrenComponent> _children = ComponentMapper.getFor(ChildrenComponent.class);
    private static ComponentMapper<CollisionComponent> _collision = ComponentMapper.getFor(CollisionComponent.class);
    private static ComponentMapper<CombatComponent> _combat = ComponentMapper.getFor(CombatComponent.class);
    private static ComponentMapper<CullComponent> _cull = ComponentMapper.getFor(CullComponent.class);
    private static ComponentMapper<InventoryComponent> _inventory = ComponentMapper.getFor(InventoryComponent.class);
    private static ComponentMapper<LayerComponent> _layer = ComponentMapper.getFor(LayerComponent.class);
    private static ComponentMapper<LightComponent> _light = ComponentMapper.getFor(LightComponent.class);
    private static ComponentMapper<ParallaxComponent> _parallax = ComponentMapper.getFor(ParallaxComponent.class);
    private static ComponentMapper<PhysicsComponent> _physics = ComponentMapper.getFor(PhysicsComponent.class);
    private static ComponentMapper<PlayerComponent> _player = ComponentMapper.getFor(PlayerComponent.class);
    private static ComponentMapper<PositionComponent> _position = ComponentMapper.getFor(PositionComponent.class);
    private static ComponentMapper<SpriteComponent> _sprite = ComponentMapper.getFor(SpriteComponent.class);
    private static ComponentMapper<StatsComponent> _stats = ComponentMapper.getFor(StatsComponent.class);


    public static BehaviorComponent behavior(Entity entity) { return _behavior.get(entity); }
    public static CameraComponent camera(Entity entity) { return _camera.get(entity); }
    public static ChildrenComponent children(Entity entity) { return _children.get(entity); }
    public static CollisionComponent collision(Entity entity) { return _collision.get(entity); }
    public static CombatComponent combat(Entity entity) { return _combat.get(entity); }
    public static CullComponent cull(Entity entity) { return _cull.get(entity); }
    public static InventoryComponent inventory(Entity entity) { return _inventory.get(entity); }
    public static LayerComponent layer(Entity entity) { return _layer.get(entity); }
    public static LightComponent light(Entity entity) { return _light.get(entity); }
    public static ParallaxComponent parallax(Entity entity) { return _parallax.get(entity); }
    public static PhysicsComponent physics(Entity entity) { return _physics.get(entity); }
    public static PlayerComponent player(Entity entity) { return _player.get(entity); }
    public static PositionComponent position(Entity entity) { return _position.get(entity); }
    public static SpriteComponent sprite(Entity entity) { return _sprite.get(entity); }
    public static StatsComponent stats(Entity entity) { return _stats.get(entity); }
}

File: core/src/bara/game/util/ComponentLookup.kt
package bara.game.util

import bara.game.ashley.ChildrenComponent
import bara.game.ashley.ParentComponent
import com.badlogic.ashley.core.ComponentMapper
import bara.game.components.*
import com.badlogic.ashley.core.Entity

object ComponentLookup {
    private val _behavior = ComponentMapper.getFor(BehaviorComponent::class.java)
    private val _camera = ComponentMapper.getFor(CameraComponent::class.java)
    private val _children = ComponentMapper.getFor(ChildrenComponent::class.java)
    private val _collision = ComponentMapper.getFor(CollisionComponent::class.java)
    private val _combat = ComponentMapper.getFor(CombatComponent::class.java)
    private val _layer = ComponentMapper.getFor(LayerComponent::class.java)
    private val _light = ComponentMapper.getFor(LightComponent::class.java)
    private val _parallax = ComponentMapper.getFor(ParallaxComponent::class.java)
    private val _parent = ComponentMapper.getFor(ParentComponent::class.java)
    private val _physics = ComponentMapper.getFor(PhysicsComponent::class.java)
    private val _player = ComponentMapper.getFor(PlayerComponent::class.java)
    private val _position = ComponentMapper.getFor(PositionComponent::class.java)
    private val _sprite = ComponentMapper.getFor(SpriteComponent::class.java)


    fun behavior(entity: Entity): BehaviorComponent? {
        return _behavior.get(entity)
    }

    fun camera(entity: Entity): CameraComponent? {
        return _camera.get(entity)
    }

    fun children(entity: Entity): ChildrenComponent? {
        return _children.get(entity)
    }

    fun collision(entity: Entity): CollisionComponent? {
        return _collision.get(entity)
    }

    fun combat(entity: Entity): CombatComponent? {
        return _combat.get(entity)
    }

    fun layer(entity: Entity): LayerComponent? {
        return _layer.get(entity)
    }

    fun light(entity: Entity): LightComponent? {
        return _light.get(entity)
    }

    fun parallax(entity: Entity): ParallaxComponent? {
        return _parallax.get(entity)
    }

    fun parent(entity: Entity): ParentComponent? {
        return _parent.get(entity)
    }

    fun physics(entity: Entity): PhysicsComponent? {
        return _physics.get(entity)
    }

    fun player(entity: Entity): PlayerComponent? {
        return _player.get(entity)
    }

    fun position(entity: Entity): PositionComponent? {
        return _position.get(entity)
    }

    fun sprite(entity: Entity): SpriteComponent? {
        return _sprite.get(entity)
    }
}
